interview-questions.md

```diff
@@ -8,50 +8,76 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+Researched answer: .map returns an array with pieces of information from the original array that the users wishes to pass in. This array will be the same as the original array. .filter returns a subset of the original array based on the criteria that the user specifies. It then decides if the array meets the criteria, and outputs the indexes which do in a new subset array.
+Source: https://thinkster.io/tutorials/100-front-end-interview-questions-challenge/map-vs-filter-vs-reduce
+
+
 
-Researched answer:
 
 1. What is iteration?
 
-Your answer:
+Your answer: Iteration is the act of iterating over something. That is, looping over something, until the certain condition is met. If the condition is not met, the function or method will continue to iterate over the thing until the condition is met. Iteration can be used with different methods; the most common method is a for loop. If one wishes to use higher order functions, one can use .map or .filter. However, .map and .filter can only iterate over arrays. A for loop can iterate over strings.
+
+Researched answer: An iterator is an object which defines a sequence; something that is specified by the user. It then returns a value based off what the user specified.
+Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators
+
+
 
-Researched answer:
 
 1. What is the difference between a function and a method?
 
-Your answer:
+Your answer: A function is a machine. That is, a function can perform a certain action, and can be used as many times as the user wishes to. A function is dynamic; this means it can take different inputs from a user, and then put these methods into use and perform the actions that the function specifies. A method can be a function, such as .map or .filter. However, methods can only be applied to certain data. a .slice() method can only be used with strings, whereas a .filter method can only be used on arrays. 
+
+Researched answer: Fucntions can execute a set of instructions on specific data or variables, and then return a result based on those instructions. Functions can be used repeatedly and eliminate the need for a developer to have to type the same code over and over again. A method is very similar to a function, however, a method must be used with another piece of data or a variable, and a function does not.
+Source: https://www.codecademy.com/article/fwd-js-methods-functions
```

Comment:
I like the fact that you mentioned built in methods but what about methods in terms of property of an object?